core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -110,15 +109,6 @@ class TaskMetrics private[spark] () extends Serializable {
    */
   def peakExecutionMemory: Long = _peakExecutionMemory.sum
 
-  /**
-   * Storage statuses of any blocks that have been updated as a result of this task.
-   */
-  def updatedBlockStatuses: Seq[(BlockId, BlockStatus)] = {
```

Comment:
Unfortunately `TaskMetrics` is a public class(via `SparkListenerTaskEnd`), so this is a breaking change.

But I do agree we should not track the updated block status, how about we still keep this method and make it always return `Nil`? We can mark it as deprecated and add comments to say that it's only here for binary compatibility.